spec/lib/booker/errors_spec.rb

```diff
@@ -69,6 +69,41 @@
   end
 end
 
+describe Booker::MidPaginationError do
+  describe 'attributes' do
+    it { expect(subject).to respond_to :error_occurred_during_params }
+    it { expect(subject).to respond_to :results_fetched_prior_to_error }
+    it { expect(subject).to respond_to :message }
+    it { expect(subject).to respond_to :error_occurred_during_params= }
+    it { expect(subject).to respond_to :results_fetched_prior_to_error= }
+    it { expect(subject).to respond_to :message= }
+  end
+
+  describe '.new' do
+    let(:error) { Booker::MidPaginationError.new(message, error_occurred_during_params, results_fetched_prior_to_error) }
+    let(:message) { "Error occurred during call mid-pagination" }
+    before do
+      expect(error.error_occurred_during_params).to eq error_occurred_during_params
+      expect(error.results_fetched_prior_to_error).to eq results_fetched_prior_to_error
+      expect(error.message).to eq message
```

Comment:
I dried these up by putting them in the before block; they're the same for both undefined/defined.